portal/app/core/assets/js/weather.js

```diff
@@ -2,24 +2,42 @@ var weather = weather || {};
 
 $(document).ready(function(ns) {
     var onLoad = function() {
-        // setup listeners
-        setupListeners();
-    },
+            // setup listeners
+            setupListeners();
+        },
 
-    setupListeners = function() {
-        $("#get-weather").on("click", function() {
-            ns.queryWeatherBasedOnZip($("#get-weather-inpt").val());
-        });
-    };
+        setupListeners = function() {
+            $("#get-weather").on("click", function() {
+                ns.queryWeatherBasedOnZip($("#get-weather-inpt").val());
+            });
+        };
 
     ns.queryWeatherBasedOnZip = function(zipCode) {
         // Query the OWM API for weather data
+        data = JSON.stringify(data, null, '\t');
```

Comment:
What is this doing? Did you test this after making this change? Where is `data` specified? 
If you run this code, it'll crash. 

I need you to figure out a way to avoid these in the future.

Reply:
I'll be more careful, fixed